### DISCUSSION TITLE
Failed to save contentItem when using "contentItem.As<LeadPart>()" in OrchardCore 2.1.3 without exceptions

### DISCUSSION THREAD

**Discussion by effapp** [OrchardCMS/OrchardCore#17270] [Q&A]
### Describe the bug

Failed to save contentItem when using "contentItem.As<LeadPart>()" in OrchardCore 2.1.3 without exceptions

```c#
var contentItem = await _contentManager.NewAsync("Lead");
var leadPart = contentItem.As<LeadPart>();
leadPart.FirstName = “FirstName”;
leadPart.LastName = “LastName”;
leadPart.Company =“ Company”;
leadPart.Email = “Email”;
leadPart.Phone = “Phone”;
leadPart.Source = “Source”;
leadPart.Status = “Status”;
leadPart.Description = “Description”;
leadPart.Apply();
await _contentManager.CreateAsync(contentItem, VersionOptions.Published);
```

The result saved to database is :

{"ContentItemId":"4we05tthw9aq15k3tt4m78j9ds","ContentItemVersionId":"4nyw8p39pywx01t6cb8kzdxz4a","ContentType":"Lead","DisplayText":"3333333333333","Latest":true,"Published":true,"ModifiedUtc":"2024-12-20T07:45:14.542097Z","PublishedUtc":"2024-12-20T07:45:14.6086407Z","CreatedUtc":"2024-12-20T07:45:14.542097Z","Owner":"431atmty3vvgf4jbc3v4qhx4cp","Author":"superadmin","TitlePart":{},"LeadPart":{"FirstName":{},"LastName":{},"Company":{},"Email":{},"Phone":{},"Source":{},"Status":{},"Description":{},"Id":null,"CreatedUtc":"2024-12-20T07:43:58Z","LastModifiedUtc":"2024-12-20T07:44:00Z"}}

The modifications made to the leadPart have not taken effect . What's the problem, please?

### Orchard Core version

version 2.1.3

**Comment by effapp** [comment]
public class LeadPart : ContentPart
{
    public string? Id { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? Company { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }

    public string? Source { get; set; }

    public string? Status { get; set; }

    public string? Description { get; set; }

    public DateTime? CreatedUtc { get; set; }

    public DateTime? LastModifiedUtc { get; set; }
}

**Comment by Piedone** [comment]
Try to follow the pattern from here: https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/a2dcc8ba17ddaacffa5b097126fb9e668256fb0c/Lombiq.TrainingDemo/Controllers/PersonListController.cs#L148
